beanstalk_worker/worker_start_scan.py

```diff
@@ -96,18 +97,43 @@ def scan_job_data(job_data):
     image_id = atomic_object.get_input_id(image_full_name)
     image_rootfs_path = os.path.join("/", image_id)
 
-    # create a directory /<image_id> where we'll mount image's rootfs
-    os.makedirs(image_rootfs_path)
 
     # configure options before mounting the image rootfs
     logger.log(level=logging.INFO,
                msg="Setting up system to mount image's rootfs")
+
+    # All these values need to be setup before creating the mount directory
+    # to handle exception in os.makedirs operation, since if the directory
+    # exists, we need to unmount it first, rmtree the directory
+    # and re-create new one and finally mount!
     mount_object = mount.Mount()
     mount_object.mountpoint = image_rootfs_path
     mount_object.image = image_id
     # mount the rootfs in read-write mode. else yum will fail
     mount_object.options = ["rw"]
 
+    try:
+        # create a directory /<image_id> where we'll mount image's rootfs
+        os.makedirs(image_rootfs_path)
+    except OSError as error:
+        logger.log(
+            level=logging.WARNING,
+            msg=str(error)
+            )
+        logger.log(
+            level=logging.INFO,
+            msg="Unmounting and removing directory %s" % image_rootfs_path)
+        mount_object.unmount()
+        shutil.rmtree(image_rootfs_path)
+        try:
+            os.makedirs(image_rootfs_path)
+        except OSError as error:
+            logger.log(
+                level=logging.CRITICAL,
+                msg=str(error)
+                )
+            raise
```

Comment:
@navidshaikh should we raise the error or we should handle it to write in logs? as if it raises the exception without handling, it will cause the worker to stop working for next jobs also.